File: new-tv-app/quickjs/src/main/assets/js/spider.js
var spider = {
    home: function(filter) { return JSON.stringify([]); },
    category: function(tid, pg, filter) { return JSON.stringify([]); },
    detail: function(ids) { return JSON.stringify([]); },
    search: function(key, quick) { return JSON.stringify([]); },
    player: function(flag, id) { return JSON.stringify({}); }
};

File: new-tv-app/catvod/src/main/java/com/github/catvod/crawler/Spider.java
package com.github.catvod.crawler;

import com.github.catvod.bean.Header;

import java.util.List;

public abstract class Spider {
    public void init() {}
    public void destroy() {}
    public String homeContent(boolean filter) throws Exception { return ""; }
    public String categoryContent(String tid, String pg, boolean filter) throws Exception { return ""; }
    public String detailContent(List<String> ids) throws Exception { return ""; }
    public String searchContent(String key, boolean quick) throws Exception { return ""; }
    public String playerContent(String flag, String id, List<Header> headers) throws Exception { return ""; }
    public boolean isVideoFormat(String url) { return false; }
}